Remove the OCF linux kernel extension instead opting to build ocf-linux
modules out of tree.
This is easier for users since no kernel config tweaking is required.
On the downside the OCF drivers can't be used, but then all of the
kernel crypto drivers are available to users via cryptosoft which is
preferred.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura ocf-linux-20120127.orig/ocf/Makefile ocf-linux-20120127/ocf/Makefile
--- ocf-linux-20120127.orig/ocf/Makefile	2013-04-27 09:27:04.413911866 -0300
+++ ocf-linux-20120127/ocf/Makefile	2013-04-27 09:27:31.131775576 -0300
@@ -2,6 +2,7 @@
 -include $(ROOTDIR)/modules/.config

 OCF_OBJS = crypto.o criov.o
+KDIR ?= /lib/modules/$(shell uname -r)/build

 ifdef CONFIG_OCF_RANDOMHARVEST
 	OCF_OBJS += random.o
@@ -78,20 +79,13 @@
 #

 ocf_make:
-	make -C /lib/modules/$(shell uname -r)/build M=`pwd` $(OCF_TARGET) CONFIG_OCF_OCF=m
-	make -C /lib/modules/$(shell uname -r)/build M=`pwd` $(OCF_TARGET) CONFIG_OCF_OCF=m CONFIG_OCF_CRYPTOSOFT=m
-	-make -C /lib/modules/$(shell uname -r)/build M=`pwd` $(OCF_TARGET) CONFIG_OCF_OCF=m CONFIG_OCF_BENCH=m
-	-make -C /lib/modules/$(shell uname -r)/build M=`pwd` $(OCF_TARGET) CONFIG_OCF_OCF=m CONFIG_OCF_OCFNULL=m
-	-make -C /lib/modules/$(shell uname -r)/build M=`pwd` $(OCF_TARGET) CONFIG_OCF_OCF=m CONFIG_OCF_HIFN=m
+	make -C $(KDIR) M=`pwd` $(OCF_TARGET) CONFIG_OCF_OCF=m CONFIG_OCF_CRYPTOSOFT=m CONFIG_OCF_BENCH=m CONFIG_OCF_OCFNULL=m

 ocf_modules:
 	$(MAKE) ocf_make OCF_TARGET=modules

 ocf_install:
 	$(MAKE) ocf_make OCF_TARGET="modules modules_install"
-	depmod
-	mkdir -p /usr/include/crypto
-	cp cryptodev.h /usr/include/crypto/.

 #
 # generate full kernel patches for 2.4 and 2.6 kernels to make patching
